homeassistant/components/sensor/dwd_weather_warnings.py

```diff
@@ -0,0 +1,243 @@
+        data['region_name'] = self._api.region_name
+
+        if self._api.region_id is not None:
+            data['region_id'] = self._api.region_id
+
+        if self._api.region_state is not None:
+            data['region_state'] = self._api.region_state
+
+        if self._api.data['time'] is not None:
+            data['last_update'] = dt_util.as_local(
+                dt_util.utc_from_timestamp(self._api.data['time'] / 1000))
+
+        if self._var_id == 'current_warning_level':
+            prefix = 'current'
+        elif self._var_id == 'advance_warning_level':
+            prefix = 'advance'
+
+        data['warning_count'] = self._api.data[prefix + '_warning_count']
+        i = 0
+        for event in self._api.data[prefix + '_warnings']:
+            i = i + 1
+
+            data['warning_{}_name'.format(i)] = event['event']
+            data['warning_{}_level'.format(i)] = event['level']
+            data['warning_{}_type'.format(i)] = event['type']
+            if len(event['headline']) > 0:
+                data['warning_{}_headline'.format(i)] = event['headline']
+            if len(event['description']) > 0:
+                data['warning_{}_description'.format(i)] = event['description']
+            if len(event['instruction']) > 0:
+                data['warning_{}_instruction'.format(i)] = event['instruction']
+
+            if event['start'] is not None:
+                data['warning_{}_start'.format(i)] = dt_util.as_local(
+                    dt_util.utc_from_timestamp(event['start'] / 1000))
+
+            if event['end'] is not None:
+                data['warning_{}_end'.format(i)] = dt_util.as_local(
+                    dt_util.utc_from_timestamp(event['end'] / 1000))
+
+        return data
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self._api.available
+
+    def update(self):
+        """Get the latest data from the DWD-Weather-Warnings API."""
+        self._api.update()
+
+
+class DwdWeatherWarningsAPI(object):
+    """Get the latest data and update the states."""
+
+    def __init__(self, region_name):
+        """Initialize the data object."""
+
+        resource = "{}{}{}?{}".format(
+            'https://',
+            'www.dwd.de',
+            '/DWD/warnungen/warnapp_landkreise/json/warnings.json',
+            'jsonp=loadWarnings'
+        )
+
+        self._rest = RestData('GET', resource, None, None, None, True)
+        self.region_name = region_name
+        self.region_id = None
+        self.region_state = None
+        self.data = None
+        self.available = True
+        self.update()
+
+    @Throttle(SCAN_INTERVAL)
+    def update(self):
+        """Get the latest data from the DWD-Weather-Warnings."""
+        try:
+            self._rest.update()
+
+            json_string = self._rest.data[24:len(self._rest.data) - 2]
+            json_obj = json.loads(json_string)
+
+            data = {}
```

Comment:
You could save a line here. My IDE complains about. ;-)

- data = {}
-
- data['time'] = json_obj['time']
+ data = {'time': json_obj['time']}

Reply:
good catch